src/util/logging/logger.go

```diff
@@ -1,172 +1,71 @@
-		Hooks:     make(logrus.LevelHooks),
-		Level:     logrus.FatalLevel,
-	}
-)
+// NewMasterLogger creates a new package-aware logger with formatting string
+func NewMasterLogger() *MasterLogger {
+	hooks := make(logrus.LevelHooks)
 
-// NewLogger creates a new modules-aware logger with formatting string
-func NewLogger(priorityKey, criticalPriority string) (logger *Logger) {
-	logger = &Logger{
+	return &MasterLogger{
 		Logger: &logrus.Logger{
 			Out: os.Stdout,
 			Formatter: &TextFormatter{
-				FullTimestamp:          true,
-				AlwaysQuoteStrings:     true,
-				QuoteEmptyFields:       true,
-				ForceFormatting:        true,
-				PriorityKey:            priorityKey,
-				HighlightPriorityValue: criticalPriority,
+				FullTimestamp:      true,
+				AlwaysQuoteStrings: true,
+				QuoteEmptyFields:   true,
+				ForceFormatting:    true,
+				DisableColors:      false,
+				ForceColors:        false,
 			},
-			Hooks: make(logrus.LevelHooks),
-			Level: logrus.InfoLevel,
-		},
-		allModulesEnabled: true,
-		moduleLoggers:     make(map[string]*Logger),
-		PriorityKey:       priorityKey,
-		CriticalPriority:  criticalPriority,
-	}
-	logger.Hooks.Add(NewModuleLogHook(""))
-	logger.moduleLoggers[""] = logger
-	return
-}
-
-// LoggerForModules creates a logger for a set of modules
-func LoggerForModules(priorityKey, criticalPriority string, enabledModules []string) *Logger {
-	logger := NewLogger(priorityKey, criticalPriority)
-	logger.allModulesEnabled = false
-	for _, moduleName := range enabledModules {
-		// Lazy instantiation
-		logger.moduleLoggers[moduleName] = nil
-	}
-	return logger
-}
-
-func (logger *Logger) cloneForModule(moduleName string) *Logger {
-	newLogger := &Logger{
-		Logger: &logrus.Logger{
-			Out:       logger.Out,
-			Formatter: logger.Formatter,
-			Hooks:     make(logrus.LevelHooks),
-			Level:     logger.Level,
+			Hooks: hooks,
+			Level: logrus.DebugLevel,
 		},
-		allModulesEnabled: logger.allModulesEnabled,
-		moduleLoggers:     logger.moduleLoggers,
-		PriorityKey:       logger.PriorityKey,
-		CriticalPriority:  logger.CriticalPriority,
 	}
-	newLogger.Hooks.Add(NewReplayHook(logger.moduleLoggers[""]))
-	newLogger.Hooks.Add(NewModuleLogHook(moduleName))
-	return newLogger
 }
 
-// MustGetLogger returns an existing logger for a given module or creates a new one
-func (logger *Logger) MustGetLogger(moduleName string) *Logger {
-	newLogger, isInCache := logger.moduleLoggers[moduleName]
-	if !(isInCache && newLogger != nil) {
-		if isInCache || logger.allModulesEnabled {
-			newLogger = logger.cloneForModule(moduleName)
-			logger.moduleLoggers[moduleName] = newLogger
-		} else {
-			newLogger = &Logger{
-				Logger:            &QuietLogger,
-				allModulesEnabled: logger.allModulesEnabled,
-				moduleLoggers:     logger.moduleLoggers,
-			}
-		}
+// PackageLogger instantiates a package-aware logger
+func (logger *MasterLogger) PackageLogger(moduleName string) *Logger {
+	return &Logger{
```

Comment:
They're all instantiated once globally per package, I don't think it would be a problem anyway, especially compared to WithField etc (which copies a map each time) or the formatter